x-pack/plugins/maps/public/react_embeddable/map_renderer.tsx

```diff
@@ -0,0 +1,117 @@
+import { MAP_SAVED_OBJECT_TYPE } from '../../common/constants';
+import { RenderToolTipContent } from '../classes/tooltips/tooltip_property';
+
+function getLayers(layerList: LayerDescriptor[]) {
+  const basemapLayer = createBasemapLayerDescriptor();
+  return basemapLayer ? [basemapLayer, ...layerList] : layerList;
+}
+
+export interface Props {
+  title?: string;
+  filters?: Filter[];
+  query?: Query;
+  timeRange?: TimeRange;
+  layerList: LayerDescriptor[];
+  mapSettings?: Partial<MapSettings>;
+  hideFilterActions?: boolean;
+  isLayerTOCOpen?: boolean;
+  mapCenter?: MapCenterAndZoom;
+  getTooltipRenderer?: () => RenderToolTipContent;
+  onApiAvailable?: (api: MapApi) => void;
+  /*
+   * Set to false to exclude sharing attributes 'data-*'.
+   */
+  isSharable?: boolean;
+}
+
+export function MapRenderer(props: Props) {
+  const mapApiRef = useRef<MapApi | undefined>(undefined);
+  const beforeApiReadyLayerListRef = useRef<LayerDescriptor[] | undefined>(undefined);
+
+  const initialState = useMemo(() => {
+    const rawState: MapSerializedState = {
+      attributes: {
+        title: props.title ?? '',
+        layerListJSON: JSON.stringify(getLayers(props.layerList)),
+      },
+      hidePanelTitles: !Boolean(props.title),
+      isLayerTOCOpen: typeof props.isLayerTOCOpen === 'boolean' ? props.isLayerTOCOpen : false,
+      hideFilterActions:
+        typeof props.hideFilterActions === 'boolean' ? props.hideFilterActions : false,
+      mapCenter: props.mapCenter,
+      mapSettings: props.mapSettings ?? {},
+      isSharable: props.isSharable,
+    };
+    if (props.getTooltipRenderer) {
+      rawState.tooltipRenderer = props.getTooltipRenderer();
+    }
+    return {
+      rawState,
+      references: [],
+    };
+    // only run onMount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (mapApiRef.current) {
+      mapApiRef.current.setLayerList(getLayers(props.layerList));
+    } else {
+      beforeApiReadyLayerListRef.current = getLayers(props.layerList);
+    }
+  }, [props.layerList]);
+
+  const parentApi = useMemo(() => {
+    return {
+      filters$: new BehaviorSubject(props.filters),
+      query$: new BehaviorSubject(props.query),
+      timeRange$: new BehaviorSubject(props.timeRange),
+    };
+    // only run onMount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  useEffect(() => {
+    parentApi.filters$.next(props.filters);
+  }, [props.filters, parentApi.filters$]);
+  useEffect(() => {
+    parentApi.query$.next(props.query);
+  }, [props.query, parentApi.query$]);
+  useEffect(() => {
+    parentApi.timeRange$.next(props.timeRange);
+  }, [props.timeRange, parentApi.timeRange$]);
+
+  return (
+    <div className="mapEmbeddableContainer">
+      <ReactEmbeddableRenderer<MapSerializedState, MapApi>
+        type={MAP_SAVED_OBJECT_TYPE}
+        state={initialState}
+        parentApi={parentApi}
```

Comment:
I think next time you merge `main` these props may need to change to something like `getParentApi={() => ...}`.

Reply:
Thanks, just merged with main and updated these. Resolved with xhttps://github.com/elastic/kibana/pull/178158/commits/301d196d4301b65ab8399d2a6219f4b486468aed